ihmc-ros2-library/src/main/java/us/ihmc/ros2/ROS2TopicNameTools.java

```diff
@@ -231,27 +234,36 @@ public static <T> T newMessageInstance(Class<T> messageType)
    @SuppressWarnings({"unchecked", "rawtypes"})
    public static <T> TopicDataType<T> newMessageTopicDataTypeInstance(Class<T> messageType)
    {
+      if (messageType.equals(Pose3D.class))
```

Comment:
Why is this needed?

Reply:
If you want to send a Pose3D as a message itself. It's not possible without this. Most of the time we don't use that. Technically to be complete we'd add the cases for the other ones in `ros2-common-interfaces/src/main/java/us/ihmc/ros2/rosidl/geometry_msgs/msg/dds`
![image](https://github.com/ihmcrobotics/ihmc-java-ros2-communication/assets/1161586/f461991e-ddea-4b02-bbf9-df19858b8ecd)

Reply:
I have filled this out for all the cases.